aiobotocore/waiter.py

```diff
@@ -1,14 +1,30 @@
 import asyncio
 
 # WaiterModel is required for client.py import
+from botocore.exceptions import ClientError
 from botocore.waiter import WaiterModel  # noqa: F401
-from botocore.waiter import Waiter, xform_name, NormalizedOperationMethod, \
-    logger, WaiterError
+from botocore.waiter import Waiter, xform_name, logger, WaiterError
 from botocore.docs.docstring import WaiterDocstring
 from botocore.utils import get_service_module_name
 
 
+class NormalizedOperationMethod:
+    def __init__(self, client_method):
+        self._client_method = client_method
+
+    @asyncio.coroutine
+    def __call__(self, **kwargs):
+        try:
+            return (yield from self._client_method(**kwargs))
```

Comment:
actually we could've had a guard to handle non-coroutine cases as well if makes sense.
thoughts @thehesiod @jettify ?

Reply:
When would there be a non coroutine case? To my understanding this triggers an aiohttp call, thus coroutine

Reply:
thought so too. have not read the whole thing to be 100% sure there are only coroutine cases. still can easily be fixed later.